### PR TITLE
Errorbars accept marker_options and follow prop_cycle

### PR DESCRIPTION
**Pull request by as691454**
#9920 #9922 
Errorbars now 

1. accept marker options, such as mfc (used to crash)
2. follows prop_cycle (used to ignore it and always draw solid line)

The first one is important for making black and white plots (clear and filled squares) - cheaper to publish than color graphs. Property cycle fix is nice for plotting several lines at once from pandas DataFrame, for instance.

~~~python
import matplotlib.pyplot as plt
plt.rc(["xtick","ytick"],labelsize=16)
plt.rc("lines", lw=3, markersize=12, markeredgewidth=3)
plt.rc("errorbar", capsize=5)
plt.rc("axes", prop_cycle="cycler(ls=['--',':'],marker=['s','s'],mfc=['k','w'])")
plt.errorbar(x=[2,4,10],y=[3,2,4],yerr=0.5)
plt.errorbar(x=[2,4,10],y=[6,4,2],yerr=0.5)
plt.show()
~~~

test

### CODE REVIEW

**Comment by jklymak** [comment]
This really needs a test...  If you need help, let us know.  http://matplotlib.org/devdocs/devel/testing.html

**Comment by jklymak** [comment]
... and probably an example, API change note...

**Comment by phobson** [comment]
PEP8 D'oh! I'll fix that and add the example + changeling entry tomorrow

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -2815,6 +2815,16 @@ def test_errobar_nonefmt():
         assert np.all(errbar.get_color() == mcolors.to_rgba('C0'))
 
 
+@image_comparison(baseline_images=['errorbar_with_prop_cycle'],
+                  extensions=['png'])
```
Comment (dstansby) [inline lib/matplotlib/tests/test_axes.py]: Can this get a `style=mpl20` argument so it uses the new style?

Comment (phobson) [reply lib/matplotlib/tests/test_axes.py]: yes. thanks for the nudge. i can't the docs to build properly on my machine at home so this has stalled out a bit.

**Comment by jklymak** [comment]
Thanks all, it was a team effort but we got it done!  (PS hope its OK that I merged - my commits only changed the style of the test to mpl20 and remove_text=True)

**Pull request merged**